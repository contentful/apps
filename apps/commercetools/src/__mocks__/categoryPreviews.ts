import { Category } from './../interfaces';

const categoryPreviews: Category[] = [
  {
    id: '28a7442d',
    name: "Mens' clothes",
    slug: 'men-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/28a7442d/general'
  },
  {
    id: '38a7442d',
    name: "Womens' clothes",
    slug: 'women-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/38a7442d/general'
  },
  {
    id: '48a7442d',
    name: "Kids' clothes",
    slug: 'kids-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/48a7442d/general'
  }
];

export default categoryPreviews;
