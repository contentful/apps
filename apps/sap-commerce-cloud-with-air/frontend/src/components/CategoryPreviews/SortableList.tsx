import * as React from 'react';
import { SortableContainer } from 'react-sortable-hoc';
import { DeleteFn, Category } from '../../interfaces';
import { SortableListItem } from './SortableListItem';

export interface Props {
  disabled: boolean;
  categoryPreviews: Category[];
  deleteFn: DeleteFn;
}

export const SortableList = SortableContainer<Props>(
  ({ disabled, deleteFn, categoryPreviews }: Props) => {
    const itemsAreSortable = categoryPreviews.length > 1;
    return (
      <div>
        {categoryPreviews.map((category, index) => {
          return (
            <SortableListItem
              disabled={disabled}
              key={`${category.id}-${category.name}`}
              category={category}
              index={index}
              onDelete={() => deleteFn(index)}
              isSortable={itemsAreSortable}
            />
          );
        })}
      </div>
    );
  }
);
