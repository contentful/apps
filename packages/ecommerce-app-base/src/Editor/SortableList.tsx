import { FC } from 'react';
import { DeleteFn, Product } from '../types';
import { SortableListItem } from './SortableListItem';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
  skuType?: string;
}

export const SortableList: FC<Props> = ({ disabled, deleteFn, productPreviews = [], skuType }) => {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
            skuType={skuType}
          />
        );
      })}
    </div>
  );
};
