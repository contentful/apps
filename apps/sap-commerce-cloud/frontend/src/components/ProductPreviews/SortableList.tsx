import { DeleteFn, Product } from '@interfaces';
import { SortableListItem } from '@components/ProductPreviews/SortableListItem';
import { styles } from '@components/ProductPreviews/SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export function SortableList({ disabled, deleteFn, productPreviews = [] }: Props) {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
}
