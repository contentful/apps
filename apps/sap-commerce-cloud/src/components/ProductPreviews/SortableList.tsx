import { FC } from 'react';
import { DeleteFn, Product } from '../../interfaces';
import { SortableListItem } from './SortableListItem';
import { styles } from './SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export const SortableList: FC<Props> = ({ disabled, deleteFn, productPreviews = [] }) => {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
};
