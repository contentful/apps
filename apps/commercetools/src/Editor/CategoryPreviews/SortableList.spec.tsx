import React from 'react';
import identity from 'lodash/identity';
import { render, cleanup } from '@testing-library/react';
import { Props, SortableList } from './SortableList';
import categoryPreviews from '../../__mocks__/categoryPreviews';

const defaultProps: Props = {
  disabled: false,
  categoryPreviews,
  deleteFn: jest.fn()
};

const renderComponent = (props: Props) => {
  return render(<SortableList {...props} />);
};

jest.mock('react-sortable-hoc', () => ({
  SortableContainer: identity,
  SortableElement: identity,
  SortableHandle: identity
}));

describe('SortableList', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const component = renderComponent(defaultProps);
    expect(component.container).toMatchSnapshot();
  });
});
