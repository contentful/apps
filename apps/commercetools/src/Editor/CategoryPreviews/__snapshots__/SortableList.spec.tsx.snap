// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableList should render successfully 1`] = `
<div>
  <div>
    <div
      class="Card__Card___xXJsT css-pivbrd Card__Card--padding-default___b6Brl"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder category
        </span>
      </div>
      <section
        class="css-ijlwe4"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Mens' clothes
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-1b5bj2l f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            men-regular
          </h2>
        </div>
      </section>
      <div
        class="css-zdzpuy"
      >
        <a
          href="https://mc.commercetools.com/example-project/categories/28a7442d/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Card__Card___xXJsT css-pivbrd Card__Card--padding-default___b6Brl"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder category
        </span>
      </div>
      <section
        class="css-ijlwe4"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Womens' clothes
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-1b5bj2l f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            women-regular
          </h2>
        </div>
      </section>
      <div
        class="css-zdzpuy"
      >
        <a
          href="https://mc.commercetools.com/example-project/categories/38a7442d/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Card__Card___xXJsT css-pivbrd Card__Card--padding-default___b6Brl"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder category
        </span>
      </div>
      <section
        class="css-ijlwe4"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Kids' clothes
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-1b5bj2l f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            kids-regular
          </h2>
        </div>
      </section>
      <div
        class="css-zdzpuy"
      >
        <a
          href="https://mc.commercetools.com/example-project/categories/48a7442d/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
